GCRCatSimInterface/CatalogClasses.py

```diff
@@ -0,0 +1,135 @@
+                       ('galacticRv', 3.1, float)]
+
+    # below are defined getter methods used to define CatSim value-added columns
+    @cached
+    def get_hasDisk(self):
+        output = np.where(self.column_by_name('SEDs/diskLuminositiesStellar:SED_9395_583:rest')>0.0, 1.0, None)
+        return output
+
+    @cached
+    def get_hasBulge(self):
+        output = np.where(self.column_by_name('SEDs/spheroidLuminositiesStellar:SED_9395_583:rest')>0.0, 1.0, None)
+        return output
+
+    @compound('internalAv', 'internalRv')
+    def get_internalDustParams(self):
+        if ('hasDisk' in self._cannot_be_null and
+            'hasBulge' in self._cannot_be_null):
+
+            raise RuntimeError('\nUnsure whether this is a disk catalog '
+                               'or a bulge catalog\n'
+                               'self._cannot_be_null %s' % self._cannot_be_null)
+        elif 'hasDisk' in self._cannot_be_null:
+            lum_type = 'disk'
+        elif 'hasBulge' in self._cannot_be_null:
+            lum_type = 'spheroid'
+        else:
+             raise RuntimeError('\nUnsure whether this is a disk catalog '
+                               'or a bulge catalog\n'
+                               'self._cannot_be_null %s' % self._cannot_be_null)
+
+        b_name = 'otherLuminosities/%sLuminositiesStellar:B:rest' % lum_type
+        b_dust_name = 'otherLuminosities/%sLuminositiesStellar:B:rest:dustAtlas' % lum_type
+
+        v_name = 'otherLuminosities/%sLuminositiesStellar:V:rest' % lum_type
+        v_dust_name = 'otherLuminosities/%sLuminositiesStellar:V:rest:dustAtlas' % lum_type
+
+        av_list = -2.5*(np.log10(self.column_by_name(v_dust_name)) -
+                        np.log10(self.column_by_name(v_name)))
+
+        ebv_list = -2.5*(np.log10(self.column_by_name(b_dust_name)) -
+                         np.log10(self.column_by_name(v_dust_name)) -
+                         np.log10(self.column_by_name(b_name)) +
+                         np.log10(self.column_by_name(v_name)))
+
+        rv_list = av_list/ebv_list
+
+        # this is a hack to replace anomalous values of dust extinction
+        # with more reasonable values
+        if not hasattr(self, '_dust_rng'):
+            self._dust_rng = np.random.RandomState(182314)
+
+        offensive_av = np.where(np.logical_or(av_list<0.001, av_list>3.1))
+        av_list[offensive_av] = self._dust_rng.random_sample(len(offensive_av[0]))*3.1+0.001
+
+        offensive_rv = np.where(np.logical_or(np.isnan(rv_list),
+                                np.logical_or(rv_list<1.0, rv_list>5.0)))
+        rv_list[offensive_rv] = self._dust_rng.random_sample(len(offensive_rv[0]))*4.0+1.0
+
+        return np.array([av_list, rv_list])
+
+    @compound('sedFilename', 'fittedMagNorm')
+    def get_fittedSedAndNorm(self):
+
+        if not hasattr(self, '_disk_flux_names'):
+            catsim_mag_file = os.path.join(getPackageDir('gcr_catalogs'),
```

Comment:
we should move `CatSimSupport` into `GCRCatSimInterface`, and then here we use `getPackageDir('GCRCatSimInterface')`